api/v0/proto/retrievability/share_service.proto

```diff
@@ -0,0 +1,101 @@
+syntax = "proto3";
+
+package celestia.v0.retrievability.share;
+
+import "api/v0/proto/retrievability/options.proto";
+import "api/v0/proto/retrievability/types.proto";
+
+message GetRequest {
+  // The block height.
+  uint64 height = 1;
+  // The index in the ODS.
+  uint32 index = 2;
+  // Common options.
+  optional celestia.v0.retrievability.options.Options options = 3;
+}
+
+message GetResponse {
+  // The share.
+  optional celestia.v0.retrievability.types.Share share = 1;
+  // The proof of share inclusion.
+  optional celestia.v0.retrievability.types.DataRootProof proof = 2;
+}
+
+message GetODSRequest {
+  // The block height.
+  uint64 height = 1;
+  // Common options.
+  optional celestia.v0.retrievability.options.Options options = 2;
+}
+
+message GetODSResponse {
+  // The list of shares.
+  // TODO: does this include tail padding or no?
+  repeated celestia.v0.retrievability.types.Share shares = 1;
+}
+
+message GetRangeRequest {
+  // The block height.
+  uint64 height = 1;
+  // The range in the ODS. Can span multiple rows.
+  celestia.v0.retrievability.types.Range range = 2;
+  // Common options.
+  optional celestia.v0.retrievability.options.Options options = 3;
+}
+
+message GetRangeResponse {
+  // The shares.
+  repeated celestia.v0.retrievability.types.Share shares = 1;
+  // The proof of share inclusion.
+  optional celestia.v0.retrievability.types.DataRootProof proof = 2;
+}
+
+message GetRowRequest {
+  // The block height.
+  uint64 height = 1;
+  // The row in the ODS.
+  uint32 row = 2;
+  // Common options.
+  optional celestia.v0.retrievability.options.Options options = 3;
+}
+
+message GetRowResponse {
+  // The shares.
+  repeated celestia.v0.retrievability.types.Share shares = 1;
+  // The proof of share inclusion.
+  optional celestia.v0.retrievability.types.DataRootProof proof = 2;
+}
+
+message GetByNamespaceRequest {
+  // The block height.
+  uint64 height = 1;
+  // The namespace.
+  bytes namespace = 2;
+  // Common options.
+  optional celestia.v0.retrievability.options.Options options = 3;
+}
+
+message GetByNamespaceResponse {
+  // The range in the ODS. Can span multiple rows.
+  optional celestia.v0.retrievability.types.Range range = 1;
```

Comment:
Is range is needed here? Seems to be duplicate, since It is already part of proof

Reply:
What if the user doesn't want a proof? In that case they won't know the range. Could argue that you don't really need to know the range if you don't have a proof.

Reply:
Fixed in 0d2fa75421d07b56bd84caafaf20be0ba9431df2